lib/media/eme_manager.js

```diff
@@ -44,7 +46,7 @@ goog.require('shaka.util.Uint8ArrayUtils');
  * @struct
  * @extends {shaka.util.FakeEventTarget}
  */
-shaka.media.EmeManager = function(parent, video, videoSource) {
+shaka.media.EmeManager = function(parent, video, videoSource, restrictions) {
```

Comment:
Let's make convert "parent" to "player" (type shaka.player.Player).  We're already passing the player as the parent, but let's have EME manager store this.player_ instead of this.restrictions_.

Then, instead of postProcessor(this.restrictions_), this.videoSource_.setRestrictions(this.restrictions_), we can say restrictions = this.player_.getRestrictions(), postProcessor(restrictions), this.player_.setRestrictions(restrictions).

That will make the postProcessor flow and the independent, application-level flow the same.